source/dub/generators/build.d

```diff
@@ -624,98 +635,226 @@ struct Watcher
 			kevent_t event;
 			EV_SET(&event, fd, EVFILT_VNODE, EV_ADD | EV_CLEAR, fflags, 0, null);
 			errnoEnforce(kevent(eventfd, &event, 1, null, 0, null) != -1);
+		} else version (Windows) {
+			paths ~= path; 
 		} else
 			static assert(0, "no file watcher event system");
 	}
 
-	void wait(Pid pid)
+	version(Windows)
 	{
-		import core.sys.posix.signal, core.sys.posix.sys.select;
-
-		// Block SIGCHLD
-		sigset_t set;
-		sigemptyset(&set);
-		sigaddset(&set, SIGCHLD);
-		sigprocmask(SIG_BLOCK, &set, null);
-
-		// Add dummy handler SIGCHLD
-		extern(C) void dummy_handler(int) {}
-		sigaction_t sa;
-		sa.sa_handler = &dummy_handler;
-		sa.sa_flags = 0;
-		sigemptyset(&sa.sa_mask);
-		sigaction(SIGCHLD, &sa, null);
-
-		if (!tryWait(pid).terminated)
+		// Recursively get more and more specific wrt first member of paths[]
+		private Path getAncestor(Path[] paths, Path accum = Path([]), int cdepth = 0) {
+			// vibe.d bug in opEquals?	accum should eventually == paths if paths.length = 1
+			// but it never does.
+			if(cdepth == 0) {
+				accum = Path(paths[0][0]);
+				return getAncestor(paths, accum, 1);
+			} else {
+				if(paths.all!(p => p.startsWith(accum)) && accum.length() < paths[0].length())// && accum != paths[0])
+					return getAncestor(paths, accum ~ paths[0][cdepth], cdepth + 1);
+				else
+					return accum.parentPath();
+			}
+		}
+		// Gets the most specific Ancestor Path a list of absolute paths:
+		private string getAncestor(string[] paths) {
+			auto ps = paths.map!(path => Path(path)).array; // lazy better?
+			return to!string(getAncestor(ps)).replace("/", "\\");
+		}
+		// returns true if watched file has changed:
+		bool processBuffer(void* buf) {
+			struct FILE_NOTIFY_INFORMATION {
+				uint NextEntryOffset;
+				uint Action;
+				uint FileNameLength;
+				wchar[0] FileName; // variable length struct
+			}
+			logInfo("Processing buffer.");
+			int nextOffset = 0;
+			FILE_NOTIFY_INFORMATION* info;
+			do {
+				info = cast(FILE_NOTIFY_INFORMATION*)(buf + nextOffset);
+				auto ws = (cast(wchar*)info.FileName)[0..info.FileNameLength/wchar.sizeof];
+				string fname = to!string(to!wstring(ws)); // TODO: make this better?
+				string fpath = joinPath(dir, fname).replace("/", "\\"); // why does vibe.d do this?
+				if(paths.canFind(fpath) && info.Action == FILE_ACTION_MODIFIED) {
+				   return true; 
+				}
+				// TODO: Check if this is actually a correct loop.
+				nextOffset += info.NextEntryOffset; // NextEntryOffset = 0 when it's the last. 
```

Comment:
It appears, `nextOffset` can be removed altogether and this line can be replaced with this:

    buf += info.NextEntryOffset;